test/test_connection.py

```diff
@@ -11,3 +13,36 @@ def test_connect_with_kwargs(self):
         client = _get_connection('foo-connection').database.client
         self.assertEqual(42, client.max_pool_size)
         self.assertEqual(10, client.min_pool_size)
+
+    def test_connect_lazily(self):
+        connect('mongodb://localhost:27017/foo',
+                'foo-connection',
+                connect=False)
+        client = _get_connection('foo-connection').database.client
+
+        class Article(MongoModel):
+            title = CharField()
+            class Meta:
+                connection_alias = 'foo-connection'
+        self.assertFalse(client._topology._opened)
+
+        self.assertEqual(Article.objects.count(), 0)
+        self.assertTrue(client._topology._opened)
```

Comment:
This is very clever but I think we can test the same thing without using the private MongoClient api. With the SDAM monitoring api instead: 
```python
from pymongo.monitoring import ServerHeartbeatListener

class HeartbeatStartedListener(ServerHeartbeatListener):
    def __init__(self):
        self.results = []

    def started(self, event):
        self.results.append(event)

    def succeeded(self, event):
        pass

    def failed(self, event):
        pass

heartbeat_listener = HeartbeatStartedListener()
connect('mongodb://localhost:27017/foo',
        'foo-connection',
         connect=False,
         event_listeners=[heartbeat_listener])
...

self.assertEqual(len(heartbeat_listener.results), 0)
self.assertEqual(Article.objects.count(), 0)
self.assertGreaterOrEqual(len(heartbeat_listener.results), 1)
```